lib/jnpr/junos/device.py

```diff
@@ -919,7 +919,7 @@ def execute(self, rpc_cmd, **kvargs):
             #    protocol: operation-failed
             #    error: device asdf not found
             # </rpc-reply>
-            if rpc_rsp_e.text is not None and rpc_rsp_e.text.strip() is not '':
+            if rpc_rsp_e.text.strip() is not '':
```

Comment:
This line looks like it accidentally rolled back the commit for issue #587 (fixed in pull request #590).